app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx

```diff
@@ -128,20 +133,38 @@ export class LaunchExecutionChart extends Component {
   };
 
   onChartClick = (d) => {
-    const { widget, getStatisticsLink } = this.props;
-    const id = getResult(widget).id;
-    const defaultParams = this.getDefaultLinkParams(id);
+    const { widget, launchFilters, getStatisticsLink } = this.props;
+
     const nameConfig = getItemNameConfig(d.id);
+    const id = getResult(widget).id;
+    let navigationParams;
+
+    if (!id) {
+      const appliedWidgetFilterId = widget.appliedFilters[0].id;
+      const activeFilter = launchFilters.filter((filter) => filter.id === appliedWidgetFilterId)[0];
+      const activeFilterId = activeFilter && activeFilter.id;
+      navigationParams = this.getDefaultParamsOverallStatisticsWidget(activeFilterId);
+    } else {
+      const link = getStatisticsLink(nameConfig.defectType.toUpperCase());
+      navigationParams = Object.assign(link, this.getDefaultParamsLaunchExecutionWidget(id));
+    }
 
-    const link = getStatisticsLink(nameConfig.defectType.toUpperCase());
-    this.props.navigate(Object.assign(link, defaultParams));
+    this.props.navigate(navigationParams);
   };
 
-  getDefaultLinkParams = (testItemIds) => ({
+  getDefaultParamsOverallStatisticsWidget = (activeFilterId) => ({
+    payload: {
+      projectId: this.props.project,
+      filterId: activeFilterId || ALL,
+    },
+    type: PROJECT_LAUNCHES_PAGE,
+  });
+
+  getDefaultParamsLaunchExecutionWidget = (id) => ({
     payload: {
       projectId: this.props.project,
-      filterId: 'all',
-      testItemIds,
+      filterId: ALL,
+      testItemIds: id,
     },
     type: TEST_ITEM_PAGE,
```

Comment:
When this widget used in the `overallStatisticsDonut` it shouldn't redirect to the step level (see the task requirements https://jiraeu.epam.com/browse/EPMRPP-38629).
This widget used in the `launchExecutionAndIssueStatistice` and as a part of `overallStatistics`.
Please provide the proper mechanism to manage this difference

Reply:
My inattention. Sorry for misleading